### ISSUE TITLE
Padding does not take effect until I start typing

### ISSUE THREAD

**Issue by andrewbanchich** [jwilm/alacritty#2226]
Which operating system does the issue occur on?

macOS 10.14.4

I have Alacritty's window padding settings set to the default of:

```
  padding:
    x: 10
    y: 10
```

When I open a new Alacritty window, there is 0 padding, but when I begin to type the padding setting goes into effect and is set to `10px`.

I've tried removing my config file and uninstalling and reinstalling Alacritty, but it still does this even with a fresh install.

**Comment by chrisduerr** [comment]
Does the prompt spawn in a window without any padding? What happens when you set the padding to something huge like `100`?

Does moving the mouse or starting a selection cause the padding to pop up?

**Comment by andrewbanchich** [comment]
@chrisduerr Correct, the window initially has no padding at all. When I set the padding to `100`, the window is larger, but the padding is still `0`.

Moving the mouse doesn't do anything, but clicking on the window does cause the padding to appear,

**Comment by chrisduerr** [comment]
I've just tested this on my macbook and I'm not able to reproduce it at all. Are you using the default window manager or a third party one?

Also which Alacritty version are you using?

**Comment by andrewbanchich** [comment]
Yes, I am using the default window manager.

```
~  λ brew cask info alacritty
alacritty: 0.2.9
https://github.com/jwilm/alacritty/
/usr/local/Caskroom/alacritty/0.2.9 (64B)
From: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/alacritty.rb
==> Name
Alacritty
==> Artifacts
Alacritty.app (App)
```

**Comment by andrewbanchich** [comment]
I also noticed that clicking _anywhere_ will cause the padding setting to take effect. If I click outside of Alacritty, onto the desktop, into another program window, it will cause it to start working.

**Comment by andrewbanchich** [comment]
0.3.0 upgrade fixed this for me. Thank you!

**Comment by andrewbanchich** [comment]
Actually, I was wrong: the only difference is that now moving the cursor over the Alacritty window does cause padding to take affect.

**Comment by INCHMAN1900** [comment]
I made a video about this problem, but it only appears when I type Chinese characters.

Link is [Here](https://youtu.be/bsOJ84x2wn8)

system: **macOS**
alacritty version: 0.3.0
padding: 2

**Comment by xinhangliu** [comment]
@INCHMAN1900 Same problem.

**Comment by andrewbanchich** [comment]
That looks different from my issue. Mine starts with 0 padding and then registers the padding after I type anything. Yours starts with padding and then removes it after typing Chinese characters.

**Comment by andrewbanchich** [comment]
Another strange thing about this: if I comment out the `padding` setting entirely, the window opens, and when I move my cursor over it I can still see the text adjust by only a pixel or so, even though there is no padding set.

However, if I set the `padding` to `0` each, it does not do this; the text stays in place 100%.

**Comment by chrisduerr** [comment]
@andrewbanchich The default padding of the latest release is `3x3`. This has been changed on master though, but that's likely what you're experiencing.

**Comment by andrewbanchich** [comment]
So I just wiped by entire MacBook's hard drive and reinstalled macOS. The first thing I did was install Homebrew, then `brew cask install alacritty`, then changed `padding` in the config file to `x: 20` and `y: 20`.

This still happens.

**Comment by chrisduerr** [comment]
I'd be very curious if the #2438 PR fixes this, however it's not possible to test this since there's still some upstream fixes missing.

Once that's done, it's probably worth re-testing. I'll also try to reproduce this on my laptop at work again, but iirc last time I tried I was not successful.

@andrewbanchich Which apple device are you running specifically? Not that it should matter much, but it might be insightful.

**Comment by andrewbanchich** [comment]
Thanks! I'm really curious what is causing this. I tried outputting logs in debug mode and the ones that appear when the padding appears are `glutin`, but I don't see anything that looks out-of-place.

I'm on macOS Mojave 10.14.5
MacBook Pro (15-inch, 2018)
Processor: 2.2 GHz Intel Core i7
Memory: 16 GB 2400 MHz DDR4
Graphics: Intel UHD Graphics 630 1536 MB

**Comment by chrisduerr** [comment]
Just fyi, I've tested this again at work and still was not able to reproduce it. Though I'm running the macOS tiling window manager chunkwm.

If anyone else is capable of reproducing this, I'd be very curious.

**Comment by kchibisov** [comment]
I've tested on friends laptop with macOs 10.14.5 (13-inch, 2018)  + touch bar and I was able to reproduce this issue with plain `bashrc`. But the one thing I've noticed, that it doesn't "bad" every time. For e.g. when launching `alacritty -e htop` every thing is fine and when launching `alacritty -e bash` also fine, but starting  from dock/spotlight (so in this case `alacritty` invokes default shell(bash) with `--login`) results in the following issue.

**Comment by andrewbanchich** [comment]
And from my own tests, it happens in bash, zsh, and fish.

**Comment by chrisduerr** [comment]
Huh, that sounds extremely odd. Though I personally never launch Alacritty from dock/spotlight so maybe that might be the only way to reproduce this? I can try that again tomorrow, though it wouldn't be surprising if this just isn't reproducible with chunkwm.

**Comment by sodiumjoe** [comment]
This happens to me too, and https://github.com/jwilm/alacritty/pull/2438 does not appear to fix it.

**Comment by chrisduerr** [comment]
Oh, thanks for bringing this up again @sodiumjoe. After updating my system, I can actually now reproduce this too.

What do you mean by `every time I saw alacritty.yml` though?

**Comment by sodiumjoe** [comment]
Sorry, every time I *save* `alacritty.yml`. ie, when alacritty restarts.

**Comment by chrisduerr** [comment]
I've got some time to look into this on my work machine, please let me know if https://github.com/jwilm/alacritty/pull/3142 fixes this.

**Comment by andrewbanchich** [comment]
Actually, for me this issue went away when I updated to macOS Catalina. Feel free to close this if you'd like.

Thanks!

**Comment by chrisduerr** [comment]
Yeah, the issue at startup has been fixed in a macOS update for me too, however there should still be an issue when the font is resized or changed in the config. That is fixed by #3142.

**Comment by sodiumjoe** [comment]
@chrisduerr confirmed fixed on macos mojave 🎉 thank you!
